runtime/libia2/include/ia2_threads.h

```diff
@@ -2,6 +2,8 @@
 
 #include "ia2.h"
 
+#if IA2_DEBUG_LOG
```

Comment:
So it seems like you both agree we shouldn't use `IA2_DEBUG_LOG` for this (https://github.com/immunant/IA2-Phase2/pull/557#discussion_r2093574063), even though it sounds pretty much exactly like what it's supposed to do.  The more defines we add, though, the more combinations we need to test and the more a user has to understand, even though these two uses are very similar.

Reply:
`IA2_DEBUG_LOG` can get very verbose if you spin up lots of threads and those messages are intertwined with whatever messages the program normally prints out to stderr/stdout. The idea here is to have a mode that doesn't add extra messages to the logs but allows you to get a better picture of what's in the address space.

Reply:
Ideally you shouldn't have to set a flag to enable the memory tracking, but since it serializes thread creation and limits the number of threads this isn't an option. It's also a good idea to do as little as possible in the TCB for something that would run in production so I think another flag here is acceptable.

Reply:
Okay, I opened #572 for this.